Common/Brokerages/KrakenBrokerageModel.cs

```diff
@@ -0,0 +1,179 @@
+                );
+
+                return false;
+            }
+            return base.CanSubmitOrder(security, order, out message);
+        }
+
+        /// <summary>
+        /// Provides Kraken fee model
+        /// </summary>
+        /// <param name="security"></param>
+        /// <returns></returns>
+        public override IFeeModel GetFeeModel(Security security)
+        {
+            return new KrakenFeeModel();
+        }
+
+        /// <summary>
+        /// Gets a new buying power model for the security, returning the default model with the security's configured leverage.
+        /// For cash accounts, leverage = 1 is used.
+        /// For margin trading, max leverage = 5
+        /// </summary>
+        /// <param name="security">The security to get a buying power model for</param>
+        /// <returns>The buying power model for this brokerage/security</returns>
+        public override IBuyingPowerModel GetBuyingPowerModel(Security security)
+        {
+            if (AccountType == AccountType.Margin)
+            {
+                foreach (var lev in CoinLeverage)
+                {
+                    if (security.Symbol.Value.StartsWith(lev.Key))
+                    {
+                        return new SecurityMarginModel(lev.Value);
+                    }
+                }
+            }
+                   
+            return new CashBuyingPowerModel();
+        }
+
+        /// <summary>
+        /// Kraken global leverage rule
+        /// </summary>
+        /// <param name="security"></param>
+        /// <returns></returns>
+        public override decimal GetLeverage(Security security)
+        {
+            if (AccountType == AccountType.Cash)
+            {
+                return 1m;
+            }
+
+            // first check whether this security support margin only with fiats.
+            foreach (var coin in _onlyFiatsAvailableMargin.Where(coin => security.Symbol.Value.StartsWith(coin)).Where(coin => _fiatsAvailableMargin.Any(rightFiat => security.Symbol.Value.EndsWith(rightFiat))))
+            {
+                return CoinLeverage[coin];
+            }
+
+            List<string> extendedCoinArray = new() {"BTC", "ETH"};
+            extendedCoinArray.AddRange(_fiatsAvailableMargin);
+            // Then check whether this security support margin with ETH.
+            foreach (var coin in _ethAvailableMargin.Where(coin => security.Symbol.Value.StartsWith(coin)).Where(coin => extendedCoinArray.Any(rightFiat => security.Symbol.Value.EndsWith(rightFiat))))
+            {
+                return CoinLeverage[coin];
+            }
+
+            extendedCoinArray.Remove("ETH");
+            // At the end check all others.
+            foreach (var coin in CoinLeverage.Keys.Where(coin => security.Symbol.Value.StartsWith(coin)).Where(coin => extendedCoinArray.Any(rightFiat => security.Symbol.Value.EndsWith(rightFiat))))
+            {
+                return CoinLeverage[coin];
+            }
```

Comment:
Cool! let's please add a nice test collection for all these different returned leverage cases 👍